tests/check-entries-csv

```diff
@@ -681,35 +784,70 @@ def check_row(row, csv):
 
 
 
-def check(scriptname, filename):
+def check(scriptname, filename,configtoml,meetpath):
     # Pretty-printer helper (so we only print the name once).
     reset_filename_state(filename)
 
     csv = oplcsv.Csv(filename)
 
+    meet_id = os.path.basename(os.path.dirname(filename))
+    meet_date = None
+
     check_headers(csv)
 
+    if meetpath != '':
+        meet = oplcsv.Csv(meetpath)
+        #Make sure that we have a proper date
+        if len(meet.rows) ==1 and "Date" in meet.fieldnames:
+            meet_date = meet.rows[0][meet.index("Date")]
+            if meet_date.count('-') != 2 or not meet_date.replace('-','').isdigit():
+                meet_date = None
+
+
+
     for row in csv.rows:
         if len(row) != len(csv.fieldnames):
             perror("Column count mismatch for row: %s" % ','.join(row))
             continue
-
-        check_row(row, csv)
+    
+        check_row(row, csv,configtoml,meet_date,meet_id)
 
 
 if __name__ == '__main__':
     # If arguments are provided, check only those files.
     if len(sys.argv) > 1:
         for f in sys.argv[1:]:
-            check(sys.argv[0], f)
+            check(sys.argv[0], f,None,None) #Need to think of a good syntax for passing config files
 
     # Otherwise, check every file named entries.csv in the dirtree,
     # which is much faster than using `find -exec`.
     else:
+        configpath = ''
+        configtoml = None
+        configfile = None
         for dirname, subdirs, files in os.walk(os.getcwd()):
             if 'entries.csv' in files:
                 entriespath = dirname + os.sep + 'entries.csv'
-                check(sys.argv[0], entriespath)
+
+
+                meetpath =''
+                if os.path.isfile(os.path.dirname(dirname)+os.sep+'CONFIG.toml'):
```

Comment:
I'm a little confused how this works... this is checking for the existence of a file named `CONFIG.toml` in the same directory as the `entries.csv` file. Would it not be in the parent directory? We should probably walk up at least a few parents...

But this is doing a bunch of checks for each entries.csv. Instead, since the tree is visited in depth-first order, couldn't we just check for the presence of `CONFIG.toml` in `files`, and if so, update it statefully and remember that? Then it would be true for all visited children. If we walk to a directory that is outside of that, it would have to be forgotten, but that's easy to check without making a syscall.